addons/mail/static/src/components/composer_suggested_partners/composer_suggested_partners.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="mail.ComposerSuggestedPartners" owl="1">
+        <!-- List of followers -->
+        <div class="o_composer_suggested_partners">
+            <t t-if="state.more" t-foreach="thread.suggestedPartners" t-as="recipient" t-key="partner_id">
+                <div t-attf-title="Add as recipient and follower (reason: {{recipient.reason}})">
+                    <div class="custom-control custom-checkbox">
+                        <input t-on-change="_onChangeSuggestedPartners(recipient_index)" type="checkbox" class="custom-control-input" t-attf-id="o_composer_suggested_partners_p{{recipient_index}}" t-att-checked="recipient.checked ? 'checked' : undefined" t-att-data-fullname="recipient.full_name"/>
+                        <label t-attf-for="o_composer_suggested_partners_p{{recipient_index}}" class="custom-control-label">
+                            <t t-esc="recipient.name"/>
+                            <t t-if="recipient.email_address">(<t t-esc="recipient.email_address"/>)</t>
+                        </label>
+                    </div>
+                </div>
+            </t>
+
+            <t t-if="!state.more" t-foreach="thread.suggestedPartners.slice(0,3)" t-as="recipient" t-key="partner_id">
+                <div t-attf-title="Add as recipient and follower (reason: {{recipient.reason}})">
+                    <div class="custom-control custom-checkbox">
+                        <input t-on-change="_onChangeSuggestedPartners(recipient_index)" type="checkbox" class="custom-control-input" t-attf-id="o_composer_suggested_partners_p{{recipient_index}}" t-att-checked="recipient.checked ? 'checked' : undefined" t-att-data-fullname="recipient.full_name"/>
+                        <label t-attf-for="o_composer_suggested_partners_p{{recipient_index}}" class="custom-control-label">
+                            <t t-esc="recipient.name"/>
+                            <t t-if="recipient.email_address">(<t t-esc="recipient.email_address"/>)</t>
+                        </label>
```

Comment:
`<label for="">` and `<input>` make sense in a `<form>`. This is not the case here, so use something else for the same intended rendering instead. (I know the previous code had this, but that's not a valid reason to do the same mistake)

Reply:
This is completely wrong.

The `<label for="">` trigger the checkbox when you click on the label, this is an accessibility requirement for a checkbox. Also the Odoo CSS remove the checkbox without label.

Reply:
> This is completely wrong.
> 
> The `<label for="">` trigger the checkbox when you click on the label, this is an accessibility requirement for a checkbox.

Ok

> Also the Odoo CSS remove the checkbox without label.

What do you mean?

Reply:
The only way you have to use a (heavy) custom style on a checkbox is to hide the system checkbox and use the label as trigger. Then you can design your check box with what ever you want (background image or like Odoo a `::before`).

If you show the system checkbox: 
![image](https://user-images.githubusercontent.com/1810149/89028086-f672e200-d32b-11ea-8ad3-0b91ee48d880.png)

Reply:
Cool, I wasn't aware of that before. From my understanding, UI elements like check/radio boxes, dropdown menu, etc. have lots of browser-specific default styling, and in order to have similar style among browsers it's best to just make another one from scratch and hide the default one.

I'm not against this approach, it makes sense as a solution to this problem. I just don't like the amount of "cryptic" classnames in template we have to add, like `custom-control-input`. I know it's the way bootstrap work: special attributes that have an impact on style and behaviour. But this is annoying because this adds too much "magic" that we don't have full control on. For instance, the magic requires a very specific layout (lots of classnames with exact parent/child depth), and if any of them are omitted, feature & style are broken.

So all in all, I don't like that we use bootstrap magic for the checkbox and would prefer our own way of doing it. It's too long to enumerate all issues of using bootstrap, so sorry for not going into the details to clarify my point. Also when I said issues with using boostrap, it's not that I have a "serious" personal issue with this library (it has lots of good stuffs). It's just that I've experienced many design issues in using bootstrap in Odoo, and the (clean) solutions consist of avoiding using boostrap (our design conflicts with their).

Realistically though... keep this in this PR, because we very likely won't change anything for a long time: too much effort, as the whole code of Odoo relies on bootstrap and this approach.

**TL;DR: interesting, have some reserves on using boostrap solution (would prefer own solution), but too ambitious so let's keep it that way.**

Reply:
> From my understanding, UI elements like check/radio boxes, dropdown menu, etc. have lots of browser-specific default styling, and in order to have similar style among browsers it's best to just make another one from scratch and hide the default one.

This is mostly true. But last time I checked, we also didn't have any CSS property to customize those elements.

> So all in all, I don't like that we use bootstrap magic for the checkbox and would prefer our own way of doing it.

You will basically rebuild the same thing since there is no alternative.

> It's just that I've experienced many design issues in using bootstrap in Odoo, and the (clean) solutions consist of avoiding using boostrap (our design conflicts with their).

I see it on the opposite way: you have a lot of issue because you customized bootstrap too much instead of using the framework to do what you want. CSS is tricky, hard to maintain, hated by devs and hard to test (browser + OS). Using bootstrap (or any other major CSS framework) mean you don't have to worry about CSS: everything that is hard and costly is already done for you.

Of course you still sometimes need to write some custom CSS. But it's should be done properly and integrated with the framework as much as possible (SCSS extend and/or variables). Bootstrap also has [guide lines](https://getbootstrap.com/docs/4.5/extend/approach/) for this.

Good CSS is hard.